stl/inc/xutility

```diff
@@ -1502,6 +1502,11 @@ using _Enable_if_execution_policy_t = typename remove_reference_t<_ExPo>::_Stand
 
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+template <class _Ty>
+concept _Allocator = _Is_allocator<_Ty>::value;
+#endif // _HAS_CXX20
+

```

Comment:
*Allocator* is a named requirements, but there isn't a concept of that in the standard.

I perviously used enable_if, but *you* suggested using concepts:)

Reply:
I don't think you need either, because the requirement on Allocator to be an actual allocator is a precondition. Yeah, that was confusing on my part though.

Reply:
Oh, I see what you concerned about. There **are** constraints in the standard.
> Constraints: `SAlloc` is a type that qualifies as an allocator ([container.requirements.general]).

`SAlloc` in standard is `_Alloc2` here. Therefore, I need to constrain the type parameters in either way.

Reply:
You're right. We thought it was odd that that's the only parameter in <sstream>. It's possible the standard shouldn't be using Constaints: here

Reply:
What should I do, then?

Reply:
leave as is. We dug into it and it looks like you did what the standard says, but the standard is a bit odd.

Someone else will go over it and then we can merge the PR

Reply:
(Reactivating comment now that I've checked and overhauled all of the constraints.)

There are now only 4 uses of this `concept`, which could be easily converted to `enable_if_t`. Since the tests are passing for EDG, I think it's okay to keep using this `concept` and consider it an early trial of #602. If we encounter any compiler errors, we can report them and change to SFINAE as a fallback.